trieste/space.py

```diff
@@ -44,6 +46,21 @@ def __contains__(self, value: TensorType) -> Union[bool, tf.Tensor]:
             from this :class:`SearchSpace`.
         """
 
+    @abstractmethod
+    def __mul__(self: Type[SP], other: SP) -> SP:
```

Comment:
i got this wrong when i first wrote it
```suggestion
    def __mul__(self: SP, other: SP) -> SP:
```